types/ckeditor__ckeditor5-engine/src/conversion/upcastdispatcher.d.ts

```diff
@@ -18,10 +17,10 @@ export interface UpcastConversionData {
 }
 
 export interface UpcastConversionApi {
-    consumable: ViewConsumable;
-    schema: Schema;
-    store: Record<string, any>;
-    writer: Writer;
+    consumable?: ViewConsumable;
```

Comment:
Making these four optional introduced a whole bunch of errors into my plugins' upcasters. I'm sure that they are guaranteed to be set in the places where I'm using them (for example, the last example [here](https://ckeditor.com/docs/ckeditor5/latest/api/module_engine_conversion_upcasthelpers-UpcastHelpers.html#function-elementToElement) -- when can they be undefined? Do we need a separate interface or something for the cases where they can be undefined? I don't think we want to force everyone to do null checks in all their upcasters that use the conversion api.

Reply:
I'm sure I made them optional based on upstream tests, that sometimes take some liberties.

I've reverted this change.